geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/records/GeoAvroRecordSetWriterFactory.scala

```diff
@@ -0,0 +1,111 @@
+package org.geomesa.nifi.datastore.processor.records
+
+import java.io.OutputStream
+import java.util
+import java.util.Collections
+
+import org.apache.nifi.annotation.documentation.{CapabilityDescription, Tags}
+import org.apache.nifi.components.PropertyDescriptor
+import org.apache.nifi.controller.AbstractControllerService
+import org.apache.nifi.expression.ExpressionLanguageScope
+import org.apache.nifi.logging.ComponentLog
+import org.apache.nifi.processor.util.StandardValidators
+import org.apache.nifi.schema.access.SchemaNameAsAttribute
+import org.apache.nifi.serialization.record.{Record, RecordSchema}
+import org.apache.nifi.serialization.{AbstractRecordSetWriter, RecordSetWriterFactory}
+import org.geomesa.nifi.datastore.processor.records.GeoAvroRecordSetWriterFactory.{GEOMETRY_COLUMNS, TYPE_NAME}
+import org.geomesa.nifi.datastore.processor.records.GeometryEncoding.GeometryEncoding
+import org.geomesa.nifi.datastore.processor.records.SimpleFeatureRecordConverter.TypeAndEncoding
+import org.locationtech.geomesa.features.avro.AvroDataFileWriter
+import org.locationtech.jts.geom.Point
+import org.opengis.feature.simple.SimpleFeatureType
+
+@Tags(Array("avro", "geoavro", "result", "set", "recordset", "record", "writer", "serializer", "row"))
+@CapabilityDescription("Writes the contents of a RecordSet as GeoAvro which AvroToPutGeoMesa* Processors can use.")
+class GeoAvroRecordSetWriterFactory extends AbstractControllerService with RecordSetWriterFactory {
+  // NB: This is the same as what the InheritSchemaFromRecord Strategy does
+  override def getSchema(map: util.Map[String, String], recordSchema: RecordSchema): RecordSchema = recordSchema
+
+  override def createWriter(componentLog: ComponentLog,
+                            recordSchema: RecordSchema,
+                            outputStream: OutputStream,
+                            map: util.Map[String, String]): GeoAvroRecordSetWriter = {
+    new GeoAvroRecordSetWriter(componentLog, recordSchema, outputStream, getConfigurationContext.getProperties)
+  }
+
+  override def getSupportedPropertyDescriptors: util.List[PropertyDescriptor] =
+    Collections.singletonList(GEOMETRY_COLUMNS)
+}
+
+object GeoAvroRecordSetWriterFactory {
+  val GEOMETRY_COLUMNS: PropertyDescriptor = new PropertyDescriptor.Builder()
+    .name("Geometry Columns")
+    .description("Comma-separated list of columns with geometries with type " +
+      "(and optionally the format, (Wkt, Wkb)).  " +
+      "Example: position:Point,line:LineString.  " +
+      "Note the first field is used as a the default geometry.")
+    .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+    .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+    .required(false).build
+
+  val TYPE_NAME: PropertyDescriptor = new PropertyDescriptor.Builder()
+    .name("SimpleFeature TypeName")
+    .description("The type name for the output.")
+    .expressionLanguageSupported(ExpressionLanguageScope.NONE)
```

Comment:
any reason not to support expression language here?

Reply:
Nope.  I'll change it to ExpressionLanguageScope.FLOWFILE_ATTRIBUTES.

Reply:
you probably don't want FLOWFILE_ATTRIBUTES - otherwise you'd need to evaluate it for every incoming flowfile

Reply:
Thanks.  That leaves.... VARIABLE_REGISTRY!

Reply:
:100: